### PR TITLE
fix: replaced field names with setter annotations

### LINKED ISSUE #1740: Adding appropriate annotations for modified field names of the models
**Actual Behaviour**

Currently some field names of models have been changed accoridng to the new api.

**Expected Behaviour**

Adding annotations to support the modified field names of models.

**Steps to reproduce it**

-

**LogCat for the issue**

-

**Screenshots of the issue**

-

**Would you like to work on the issue?**

Yes.

**Target Milestone to solve the issue**

https://github.com/fossasia/open-event-android/milestone/31

### PR DESCRIPTION
**Pull request by arp95**
Fixes #1740 

Changes: 
replaced field names with the setter and getter annotations. will remove the old model ones, once the new api is up.

Screenshots for the change: 
-

@iamareebjamal @Shailesh351 @shreyasnbhat @rohanagarwal94 
please review.

### CODE REVIEW

File: android/app/src/main/java/org/fossasia/openevent/data/Event.java [hunk 1]
```diff
@@ -16,77 +16,84 @@
...
     private Double latitude;
-    @JsonProperty("organizer_description")
     private String organizerDescription;
     private String identifier;
     private String type;
     private String email;
-    @JsonProperty("code_of_conduct")
     private String codeOfConduct;
 
+    @JsonGetter("event_url")
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/data/Event.java]: No need to annotate getters, prefer JsonProperty there. Getter annotation creates unnecessary boilerplate. Only annotate setters when there are more than one attribute name for same property. JsonProperty should be preferred over JsonSet/Getter. But we have to delegate to setter when there are more than one attribute name, but there is no need of getter annotation as we don't serialize the POJOs, only deserialize

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/data/Event.java (hunk 1 above)
Comment (arp95) [reply android/app/src/main/java/org/fossasia/openevent/data/Event.java]: but if the atttribute has both a setter and a getter than should i make the json property the old one which will be used for the getter and update the setter with multiple field names?

**Comment by arp95** [comment]
@iamareebjamal 
made changes.

File: android/app/src/main/java/org/fossasia/openevent/data/Event.java (hunk 1 above)
Comment (iamareebjamal) [reply android/app/src/main/java/org/fossasia/openevent/data/Event.java]: If there is a conflict in property name, then remove JsonProperty and create 2 setters with names from both properties

**Review by iamareebjamal — APPROVED** [review]

**Comment by iamareebjamal** [comment]
There are more changes to be done I guess @arp95 ?

**Comment by arp95** [comment]
@iamareebjamal 
i went through twice. but didnt find any setter method having conflicting attribute names. are there according to you?

**Comment by iamareebjamal** [comment]
I mean there are much more additions to do make it compatible with new API

**Comment by arp95** [comment]
you mean the new attributes? plus i havent changed the attributes with getters for now. will cahnge them when the new api is set up.

**Comment by iamareebjamal** [comment]
Please do those changes as well as it won't concern the old API and only make things easier when the new API is deployed

Your doc has listed some changes pertaining to attribute name in sessions, sponsors, etc. Do add setters for ambiguous attributes

https://docs.google.com/document/d/1caaCeOOTIxQn2HR1Uj9r-cHXQH5hSYzzzCJpt3rkl38/edit

**Comment by arp95** [comment]
so i have to these two things right?

add new attributes with jsonproperty and 
add setters for all conflicted attributes.

right?

**Comment by iamareebjamal** [comment]
Just the second part in this PR. First part will require migrations too

**Review by iamareebjamal — CHANGES_REQUESTED** [review]
Add changes for conflicting attribute names

**Comment by arp95** [comment]
made changes. please see now.

**Comment by iamareebjamal** [comment]
@arp95 The build is failing, please take a look

**Comment by arp95** [comment]
@iamareebjamal 
please review now.

**Review by iamareebjamal — DISMISSED** [review]

**Comment by arp95** [comment]
@Shailesh351 @shreyasnbhat @rohanagarwal94 
please review.

File: android/app/src/main/java/org/fossasia/openevent/data/Sponsor.java [hunk 2]
```diff
@@ -37,6 +36,26 @@ public String getUrl() {
         return url;
     }
 
+    @JsonSetter("session_type")
```
Comment (Shailesh351) [inline android/app/src/main/java/org/fossasia/openevent/data/Sponsor.java]: It should be `sponsor_type`

File: android/app/src/main/java/org/fossasia/openevent/data/Sponsor.java [hunk 3]
```diff
@@ -37,6 +36,26 @@ public String getUrl() {
...
+    public void setSponserType(String sponserType) {
+        this.sponsorType = sponserType;
+    }
+
+    @JsonSetter("session-type")
```
Comment (Shailesh351) [inline android/app/src/main/java/org/fossasia/openevent/data/Sponsor.java]: here also

**Review by Shailesh351 — CHANGES_REQUESTED** [review]

**Comment by arp95** [comment]
@iamareebjamal @Shailesh351 @shreyasnbhat @rohanagarwal94 
please review now.

**Review by Shailesh351 — APPROVED** [review]

**Comment by arp95** [comment]
@mariobehling @the-dagger 
please review now.

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**